lib/Newsletter/NewsletterSaveController.php

```diff
@@ -253,4 +244,34 @@ private function updateSegments(NewsletterEntity $newsletter, array $segments) {
 
     $this->entityManager->flush();
   }
+
+  private function updateOptions(NewsletterEntity $newsletter, array $options) {
```

Comment:
We don't handle orphaned newsletter options fields here. I guess it is because the list of options is defined by $optionFields so the removal of orphaned newsletter options should be handled when someone removes an option field. Am I correct?

Reply:
We don't delete any options here but I'm adding Doctrine's orphan removal for options too.